packages/dart_frog_gen/lib/src/build_route_configuration.dart

```diff
@@ -26,13 +26,45 @@ RouteConfiguration buildRouteConfiguration(Directory directory) {
   final middleware = <MiddlewareFile>[
     if (globalMiddleware != null) globalMiddleware
   ];
+  final endpoints = <String, List<RouteFile>>{};
   final routes = <RouteFile>[];
   final directories = _getRouteDirectories(
-    routesDirectory,
+    directory: routesDirectory,
+    routesDirectory: routesDirectory,
     onRoute: routes.add,
     onMiddleware: middleware.add,
+    onEndpoint: (endpoint, file) {
+      if (!endpoints.containsKey(endpoint)) {
+        endpoints[endpoint] = [file];
+      } else {
+        endpoints[endpoint]!.add(file);
+      }
+    },
   );
   final publicDirectory = Directory(path.join(directory.path, 'public'));
+
+  final conflictingEndpoints =
+      endpoints.entries.where((entry) => entry.value.length > 1);
+
+  if (conflictingEndpoints.isNotEmpty) {
+    final buffer = StringBuffer()..writeln('Conflicting paths detected.');
+
+    for (final conflict in conflictingEndpoints) {
+      final originalFilePath = path.normalize(
+        path.join('routes', conflict.value.first.path),
+      );
+      final conflictingFilePath = path.normalize(
+        path.join('routes', conflict.value.last.path),
+      );
+
+      buffer.writeln(
+        '''path "${conflict.key}" from route "$originalFilePath" conflicts with "$conflictingFilePath".''',
+      );
```

Comment:
Shall we cover the possible (possible?) scenario where the number of conflicts is >2?

Reply:
possible three way conflict scenario (posix):

```
├── conflict
│   ├── geez
│   │   └── index.dart
│   └── geez.dart
└── conflict\geez.dart
```

Reply:
Yeah good call. I’m refactoring the proposed implementation currently and will incorporate all conflicts 👍